backend/baseclass.pm

```diff
@@ -315,6 +315,9 @@ sub start_encoder ($self) {
     my $cwd = Cwd::getcwd;
     my @cmd = (qw(nice -n 19), "$bmwqemu::scriptdir/videoencoder", "$cwd/video.ogv");
     push(@cmd, '-n') if $bmwqemu::vars{NOVIDEO} || ($has_external_video_encoder_configured && !$bmwqemu::vars{EXTERNAL_VIDEO_ENCODER_ADDITIONALLY});
+    my $xres = $bmwqemu::vars{XRES} // '1024';
```

Comment:
Even if it's just the default value now we have three places with the resolution default values set, here as a string, later as integer. This is also likely the reason why the variable documentation test complains about needing it for both "common" as well as "qemu" but we should also not duplicate in the documentation. As we need the resolutions already here and not only for qemu I would say make it a property of the class itself in https://github.com/os-autoinst/os-autoinst/pull/2070/files#diff-ee84fe6541d1e2e971e9e0d4bf0187272df8fa4ac878d033b1c8c600325e9f6dR61 and set the default there and use the same variable within backend/qemu.pm . As "backend::qemu" *is* a "backend:baseclass" you just have the variable available there directly as well.

Reply:
I think I've managed to do what you've suggested here in the latest commit.